lib/sdf/link.rb

```diff
@@ -18,5 +18,45 @@ def each_sensor
         def pose
             Conversions.pose_to_eigen(xml.elements["pose"])
         end
+
+        # The link's inertial.
+        #
+        # @return Inertial struct
+        Inertial = Struct.new(:pose, :mass, :inertia)
+        def inertial
+            m = ixx = iyy = izz = 1
+            ixy = ixz = iyz = 0
+            pose = Eigen::Isometry3.new
+            if xml.elements["inertial"].respond_to?(:text)
```

Comment:
Everytime you see layers upon layers of ifs ... that should make your eyes bleed.

- In this library, we assume that the SDF is well formed. You therefore have to test for an element's presence only if it is optional in the SDF spec
- rexml elements all respond to #text. The only thing you need is to check whether it is present, e.g.

```
if inertial = xml.elements['inertial']
end
```

In general, one uses #respond_to? only for some form of "weak typing". What you actually test here for is presence/absence.

- you do the exact same pattern 7 times (check if an element is present, if it does, convert its contents to Float. Create a helper method for it. The best would probably be

```
# @api private
#
# Read an element whose text can be interpreted as float
#
# @param [REXML::Element] the element whose child is looked for
# @param [String] xpath the xpath of the child element
# @param [Float] default the default value to be returned if the element is not present
def read_float_child_element(element, xpath, default)
end
```